### PR TITLE
feat: support sha256 format image

### PR DESCRIPTION
**Pull request by Desiki-high**
ref: #249.

Convert source image such as `repo:tag@sha256xxxxxx` or `repo@sha256xxxxxx`.

### CODE REVIEW

File: pkg/adapter/rule.go [hunk 1]
```diff
@@ -34,7 +35,8 @@ const (
 // Source: 192.168.1.1/nginx:latest
 // Target: 192.168.1.1/nginx:latest-suffix
 func addSuffix(ref, suffix string) (string, error) {
-	named, err := docker.ParseDockerRef(ref)
+	re := regexp.MustCompile(`@sha256.*`)
```
Comment (imeoer) [inline pkg/adapter/rule.go:38]: Can we write a test case for this? It converts the digested reference format into `repo:latest`?
Maybe better to handle this in `if _, ok := named.(docker.Digested); ok {`.

Comment (Desiki-high) [reply pkg/adapter/rule.go]: Add test case in `rule_test.go`.

File: pkg/adapter/rule.go [hunk 2]
```diff
@@ -39,7 +40,11 @@ func addSuffix(ref, suffix string) (string, error) {
 		return "", errors.Wrap(err, "invalid source image reference")
 	}
 	if _, ok := named.(docker.Digested); ok {
-		return "", fmt.Errorf("unsupported digested image reference: %s", named.String())
+		re := regexp.MustCompile(`@sha256.*`)
```
Comment (imeoer) [inline pkg/adapter/rule.go:43]: Better to handle with `named` instead of regexp.

Comment (Desiki-high) [reply pkg/adapter/rule.go]: replace by `docker.TrimNamed(named)`.

**Review by imeoer — APPROVED** [review]
Thanks, LGTM!

**Pull request merged**